File: backend/internal/models/ItemEvent.go
package models

type ItemEvent struct {
	ID      uint   `gorm:"primaryKey" json:"id"`
	Name    string `json:"name" required:""`
	UserID  uint   `json:"user_id" gorm:"not null"`
	EventID uint   `json:"event_id" gorm:"not null"`
}

type ItemEventAdd struct {
	ID      uint   `gorm:"primaryKey" json:"id"`
	Name    string `json:"name" gorm:"not null"`
	Email   string `json:"email" gorm:"not null"`
	EventID uint   `json:"event_id" gorm:"not null"`
}

type ItemEventResponse struct {
	ID        uint   `gorm:"primaryKey" json:"id"`
	Name      string `json:"name" required:""`
	UserID    uint   `json:"user_id" gorm:"not null"`
	EventID   uint   `json:"event_id" gorm:"not null"`
	Firstname string `json:"firstname" required:""`
	Lastname  string `json:"lastname" required:""`
}

File: client/lib/core/models/itemEvent.dart
class ItemEvent {
  final int id;
  final String name;
  final int userId;
  final int eventId;
  final String firstname;
  final String lastname;

  ItemEvent({
    required this.id,
    required this.name,
    required this.userId,
    required this.eventId,
    required this.firstname,
    required this.lastname,
  });

  factory ItemEvent.fromJson(Map<String, dynamic> json) {
    return ItemEvent(
      id: json['id'],
      name: json['name'],
      userId: json['user_id'],
      eventId: json['event_id'],
      firstname: json['firstname'],
      lastname: json['lastname'],
    );
  }
}

